### DISCUSSION TITLE
Upgrade to .NET 8

### DISCUSSION THREAD

**Discussion by staterium** [Azure/azure-functions-durable-extension#2676] [Q&A] [answered]
I used the Visual Studio Upgrade Assistant to upgrade my functions project from .NET 6 to .NET 8. The project uses functions runtime v4 (in-process model), and contains many different function types, including durable functions.

Because .NET 8 does not yet support in-process, the assistant changes my process model to worker. So far so good. But then it also removes references to Microsoft.Azure.WebJobs.Extensions.DurableTask and replaces it with Microsoft.Azure.Functions.Worker.Extensions.DurableTask. This breaks my build because there is (for example) no definition for ```IDurableActivityContextt``` in the latter. 

Am I missing something?

**Comment by cgillum** [comment]
Indeed, there are some additional changes you'll need to make to your code to get it fully running. You can find a migration guide here: https://learn.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-dotnet-isolated-overview.

For `IDurableActivityContext`, I think there currently isn't any equivalent. Can you say more about how you're currently using it?

/cc @jviau

**Reply by staterium** [reply]
I use Durable Functions in the manner described in this article: https://learn.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-bindings

So for example, I have a function that is an activity, which receives an IDurableActivityContext object as a parameter, and that paramater has an ```[ActivityTrigger]``` attribute. Here is one such function:

```
[FunctionName("GetEmailInvoicesToSend_Activity")]
public async Task<List<EmailInvoicesToSend>> GetEmailInvoicesToSendAsync([ActivityTrigger] IDurableActivityContext context)
{
    var invoiceRunId = context.GetInput<int>();
    var invoiceRun = await GetInvoiceRunWithArtifactsAsync(invoiceRunId);

    var artifacts = invoiceRun
        .InvoiceRunArtifacts
        .Where(IsInvoiceArtifact);

    return artifacts
        .Select(w => new EmailInvoicesToSend
        {
            ArtifactId = w.ArtifactId.Value,
            CustomerId = w.CustomerId,
        })
        .ToList();
}
```

**Reply by ltouro** [reply, marked as answer]
if only input is needed, it's possible to bind it directly to the method signature:

```
[FunctionName("GetEmailInvoicesToSend_Activity")]
public async Task<List<EmailInvoicesToSend>> GetEmailInvoicesToSendAsync([ActivityTrigger] int invoiceRunId)
{
    var invoiceRun = await GetInvoiceRunWithArtifactsAsync(invoiceRunId);

    var artifacts = invoiceRun
        .InvoiceRunArtifacts
        .Where(IsInvoiceArtifact);

    return artifacts
        .Select(w => new EmailInvoicesToSend
        {
            ArtifactId = w.ArtifactId.Value,
            CustomerId = w.CustomerId,
        })
        .ToList();
}
```
The article mentions _In the .NET-isolated worker, only serializable types representing your input are supported for the [ActivityTrigger]._ So I guess you cannot depend on context in isolated-mode.
